### PR TITLE
[DOCS] Clarify copy for terms enum API's `complete` flag

### LINKED ISSUE #76065: [DOCS] Terms enum API's `complete` flag copy is unclear
**Elasticsearch version** (`bin/elasticsearch --version`): 7.14

**Plugins installed**: []

**JVM version** (`java -version`): N/A

**OS version** (`uname -a` if on a Unix-like system): N/A

**Description of the problem including expected versus actual behavior**:

**Steps to reproduce**:

On the website Documentation of Terms enum API (https://www.elastic.co/guide/en/elasticsearch/reference/7.x/search-terms-enum.html) stats following after API response.

The "complete" flag is false if time or space constraints were met and the set of terms examined was not the full set of available values.

While it should have been (Notice the **not** , and its very critical as it changes the whole meaning) 

The "complete" flag is false if time or space constraints were not met and the set of terms examined was not the full set of available values. 

**Provide logs (if relevant)**:

### PR DESCRIPTION
**Pull request by amitmbm**
Fixing the https://github.com/elastic/elasticsearch/issues/76065  (Docs bug on new terms enum API)

### CODE REVIEW

**Comment by amitmbm** [comment]
@esdocs

File: docs/reference/search/terms-enum.asciidoc
Comment (jrodewig) [inline docs/reference/search/terms-enum.asciidoc:39]: Thanks for raising this @amitmbm. I agree that the current wording in ambiguous. In particular, I'm not sure what "space" constraints would mean. It seems that this would most likely occur due to a timeout.

I've suggested some changes to clarify:
```suggestion
If the `complete` flag is `false`, the returned `terms` set may be incomplete.
This typically happens if the request times out.
```

However, I'd like to get @markharwood's review before merging these changes in.

**Comment by jrodewig** [comment]
@elastimachine ok to test

File: docs/reference/search/terms-enum.asciidoc
Comment (markharwood) [reply docs/reference/search/terms-enum.asciidoc:39]: The flag is intended to cover any number of scenarios where we might want the user to "try anyway" i.e. keep on typing the word that is in their head and search for it even if they don't see the word offered in the completion suggestions list.

The reasons for a term not being in the results could be varied e.g. a timeout or an error on one of the nodes. In future we might want to sort by popularity rather than alphabetically and this will introduce some data-gathering size limits that could lead to missing data the same way `terms` aggs can be less than accurate today.

Rather than users having to check a multitude of flags and conditions the thinking was this would be one flag which would help say if the suggestions were exhaustive regardless of the details of why that may be so.

Comment (jrodewig) [reply docs/reference/search/terms-enum.asciidoc:39]: Thanks @markharwood. I made some adjustments to the suggestion based on your feedback. My hope is that we can be a little more specific while still future-proofing this for the `terms`-agg-like size constraints you mentioned. Let me know what you think!

**Comment by amitmbm** [comment]
Thanks @jrodewig for reviewing the changes, and for your suggestion.
I've added the suggestion, please approve the PR, so that it can be merged.

**Comment by jrodewig** [comment]
Thanks @amitmbm. I'd still like to wait for @markharwood's approval before merging.

**Comment by amitmbm** [comment]
> Thanks @amitmbm. I'd still like to wait for @markharwood's approval before merging.

Sure, I am pinging him to that it can be merged soon, it's a new feature released in 7.14, and like me a lot of folks would be reading the docs 😄 
@markharwood Could you please take a look at the latest changes?

**Review by markharwood — APPROVED** [review]
The sizing constraints were a potential future cause if we choose to add support for sorting by popularity. I'd drop that phrase for now and leave the other two as examples e.g.

> This can occur due to a few reasons such as a request timeout or a node error.

Otherwise LGTM

**Comment by amitmbm** [comment]
> The sizing constraints were a potential future cause if we choose to add support for sorting by popularity. I'd drop that phrase for now and leave the other two as examples e.g.
> 
> > This can occur due to a few reasons such as a request timeout or a node error.
> 
> Otherwise LGTM

Thanks @markharwood  , Added your suggested changes, Could you please merge the changes.

File: docs/reference/search/terms-enum.asciidoc
Comment (jrodewig) [inline docs/reference/search/terms-enum.asciidoc:39]:
```suggestion
and should be treated as approximate. This can occur due to a few reasons, such
as a request timeout or a node error.
```

**Review by jrodewig — APPROVED** [review]
LGTM. I'll merge and backport this once the CI tests pass.

**Pull request merged**
